lib/perl/Genome/SoftwareResult/User.pm

```diff
@@ -39,5 +43,86 @@ class Genome::SoftwareResult::User {
     doc => 'links a software result to other entities which depend on it',
 };
 
+sub with_registered_users {
+    my $class = shift;
+    my %params = Params::Validate::validate(
```

Comment:
I would personally vote for `allow_extra => 1` to be set here instead of for the whole module.